src/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj

```diff
@@ -18,36 +24,77 @@
    (schema/optional-key :enabled)   schema/Bool
    (schema/optional-key :reporters) ReportersConfig})
 
-(def MetricsServiceContext
+(def RegistryContext
   {:registry (schema/maybe MetricRegistry)
-   (schema/optional-key :jmx-reporter) JmxReporter})
+   :jmx-reporter (schema/maybe JmxReporter)})
+
+(def MetricsServiceContext
+  {:registries (schema/atom {schema/Any RegistryContext})})
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Private
 
-(defn jmx-reporter
-  [registry jmx-config]
-  (doto (-> (JmxReporter/forRegistry registry)
-          (.build))
-    (.start)))
+(schema/defn jmx-reporter :- JmxReporter
+  [registry :- MetricRegistry
+   domain :- (schema/maybe schema/Str)]
+  (let [b (JmxReporter/forRegistry registry)]
+    (when-let [^String d domain]
+      (.inDomain b d))
+    (.build b)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
-(schema/defn initialize :- MetricsServiceContext
-  [config :- MetricsConfig]
-  (let [enabled-given   (contains? config :enabled)
-        jmx-config      (get-in config [:reporters :jmx])]
-    (when enabled-given
-      (log/warn "Metrics are now always enabled.  To suppress this warning remove metrics.enabled from your configuration."))
-    (let [registry (MetricRegistry.)]
-      (cond-> {:registry registry}
+(schema/defn initialize :- RegistryContext
+  [config :- MetricsConfig
+   domain :- (schema/maybe schema/Str)]
+  (let [jmx-config (get-in config [:reporters :jmx])
+        registry (MetricRegistry.)]
+    (when (contains? config :enabled)
+      (log/warn (str "Metrics are now always enabled.  "
+                     "To suppress this warning remove "
+                     "metrics.enabled from your configuration.")))
+    {:registry registry
+     :jmx-reporter (when (:enabled jmx-config)
+                     (doto ^JmxReporter (jmx-reporter registry domain)
+                       (.start)))}))
 
-        (:enabled jmx-config)
-        (assoc :jmx-reporter (jmx-reporter registry jmx-config))))))
+(schema/defn get-or-initialize! :- RegistryContext
+  [config :- MetricsConfig
+   {:keys [registries] :as context} :- MetricsServiceContext
+   registry-key :- schema/Keyword
+   domain :- schema/Str]
+  (if-let [metric-reg (get-in @registries [registry-key])]
+    metric-reg
+    (let [reg-context (initialize config domain)]
+      (swap! registries assoc registry-key reg-context)
+      reg-context)))
 
-(schema/defn stop :- MetricsServiceContext
-  [context :- MetricsServiceContext]
+(schema/defn stop :- RegistryContext
+  [context :- RegistryContext]
   (if-let [jmx-reporter (:jmx-reporter context)]
     (.close jmx-reporter))
   context)
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Comidi
+
+(defn build-handler [path]
+  (comidi/routes->handler
+   (comidi/wrap-routes
+    (comidi/context path
+        (comidi/context "/v1"
+            (comidi/context "/mbeans"
```

Comment:
Think that might be a code alignment problem :) both the `GET`s fall under this context